Loadtest dev ops 2598

FelixBlauza opened the pull request:
Code Redesign
https://ticketsystem.hpi-schul-cloud.org/browse/OPS-2598

Comment the Code
https://ticketsystem.hpi-schul-cloud.org/browse/OPS-2577

Lernstore implemented
https://ticketsystem.hpi-schul-cloud.org/browse/OPS-2513

Commit:
Make the Loadtest more reuseble

Commit:
Lernstore Module is now working

Commit:
Comments added

Commit:
Comments

Commit:
[OPS-2577] Added comments.

Commit:
[OPS-2598] Code redesign.

Commit:
[OPS-2598] Code redesign

Commit:
[OPS-2598] Code redesign

Commit:
[OPS-2598] Code redesign: frequently occuring code-snippets summarized

- Added method for failed requests
- Added header-builder for frequently occuring requests
- Added comments
- Code apperance improvement

Commit:
[OPS-2598] Code redesign

Commit:
[OPS-2598] Code redesign: minor bug fixes and added clean-up method

- Fixed on_start and on_stop bug on docTaskSet
- Added cleanUpLoadtest method, which deletes all documents, courses and teams after loadtest
- Added checkRequestMethod which checks, if a provided url is accessable
- Added comments

Commit:
[OPS-2598] Fix Errors

Commit:
[OPS-2598] Fix Errors

Commit:
[OPS-2598] Searching for bug 'team delete'

Commit:
[OPS-2598] Changed one line in loginout

Commit:
[OPS-2598] Fix Error

Commit:
[OPS-2598] Fix Error

Commit:
[OPS-2598] Added comments.

Commit:
[OPS-2598] Fix Error and added Comments

Commit:
[OPS-2598/2577] Updated readme.me

Commit:
[OPS-2598] Added method from OPS-2471

Commit:
[OPS-2598] Added method from OPS-2471

Commit:
[OPS-2598] Outsourcing method from OPS-2471

FelixBlauza committed:
Update README.md

FelixBlauza committed:
Update README.md

Commit:
[OPS-2598] Added comment

FelixBlauza committed:
Update README.md

FelixBlauza committed:
Update README.md

Commit:
[OPS-2598] Finished

dimapin commented on bbbTaskSet.py:
```diff
@@ -0,0 +1,153 @@
+    Task-Set which contains all test-tasks for working with BBB on the SchulCloud.
+
+    Args:
+        bBBKey (str) : key for using BBB
```
No context here and not clear how they are used

mrtn-nm replied:
Deletec because the variables are availible on the constant.py-file now

dimapin commented on bbbTaskSet.py:
```diff
@@ -0,0 +1,153 @@
+            timestamp = str(time.time())
+            # Creates a BBB-Room with a password
+            operator = "create"
+            urlparam = "meetingID=loadtest-" + timestamp + str(counterfirst) + "&name=loadtest-" + str(time.time()) + str(counterfirst) + "&moderatorPW=123&attendeePW=456&lockSettingsDisableMic=true"
```
Where does "&moderatorPW=123&attendeePW=456&lockSettingsDisableMic=true" come from?

FelixBlauza replied:
Random Information, which ware needed to separate moderators and normal attendees.

dimapin commented on constant.py:
```diff
@@ -0,0 +1,22 @@
+class constant():
+    wait_time = between(5, 15) # Provides a random number which will be used as waiting time for the users
+    tasks = {bbbTaskSet:1, scTaskSet:3, docTaskSet:1, reqWithoutUserTaskSet:1} # Conatins all task-sets which will be applied on the users
+    matrixMessengerHost = os.environ.get("MMHOST")
```
Please write MATRIXMESSENGER or MATRIX_MESSENGER

dimapin commented on constant.py:
```diff
@@ -0,0 +1,22 @@
+    returncodeCreated = 201 # Returncode if a Request to create something was successful
+    bBBKey = os.environ.get("BBBKEY")
+    bBBHost = os.environ.get("BBBHOST")
+    numberRooms = 3 #int(os.environ.get("BBBNUMBERROOMS"))
```
Why is the parametrization via environment variable disabled?

mrtn-nm replied:
Enabled enviroment parametrization

dimapin commented on constant.py:
```diff
@@ -0,0 +1,22 @@
+    bBBHost = os.environ.get("BBBHOST")
+    numberRooms = 3 #int(os.environ.get("BBBNUMBERROOMS"))
+    numberUsers = 6 #int(os.environ.get("BBBNUMBERUSERS"))
+    urlReqWithoutUser = "niedersachsen.cloud" # Used for reqWithoutUserTaskSet as url
```
Why is this hardcoded?

mrtn-nm replied:
Removed because no longer neccessary. 'reqWithoutUser' further will use the host domain to make reqests on /version, api./version and /nuxtversion.

dimapin commented on docTaskSet.py:
```diff
@@ -0,0 +1,251 @@
+
+        logout(self)
+
+    #@tag('test')
```
please remove

FelixBlauza replied:
Is deleted

dimapin commented on loginout.py:
```diff
@@ -0,0 +1,88 @@
+                login_post_response.failure("Failed! (username: " + self.user.login_credentials["email"] + ", http-code: "+str(login_post_response.status_code)+", header: "+str(login_post_response.headers)+")")
+            else:
+                response_header = login_post_response.headers
+                self.bearer_token = (response_header["set-cookie"]).split(";")[0].replace("jwt=", "")
```
Some explanation, what is happening here and why.

FelixBlauza replied:
Commented in the Code

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+        if url.endswith(".js"):
+            resource_urls.add(url)
+
+    for res in soup.find_all(href=True):
```
Svg and font files are missing here

mrtn-nm replied:
Added .svg and fonts

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+        if response.status_code != constant.constant.returncodeNormal:
+            response.failure(requestFailureMessage(self, response))
+
+def lernStore(self):  
```
Please split this method. 'and' is a sign, that the method should be splitted.

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+
+            # Request to the Lernstore to get the internal id of the course
+            with self.client.request("GET",
+                "https://api.staging.niedersachsen.hpi-schul-cloud.org/lessons?courseId=" + courseId,
```
Hard coded target

FelixBlauza replied:
Changed to a more flexible url.

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+
+                # Adding a material from the Lernstore to the course
+                with self.client.request("POST",
+                    "https://api.staging.niedersachsen.hpi-schul-cloud.org/lessons/" + courseId_Lernstore + "/material",
```
Hard coded url

FelixBlauza replied:
Changed to a more flexible url.

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+                "sec-fetch-site"    : "same-origin",
+                "x-requested-with"  : "XMLHttpRequest"
+            },
+            data = "privacy_permission=anonymous&openNewTab=true&name=bettermarks&url=https://acc.bettermarks.com/Fv1.0/schulcloud/de_ni_staging/login&key=&logo_url=https://acc.bettermarks.com/app/assets/bm-logo.png&isLocal=true&resource_link_id=&lti_version=&lti_message_type=&isTemplate=false&skipConsent=false&createdAt=2021-01-14T13:35:44.689Z&updatedAt=2021-01-14T13:35:44.689Z&__v=0&originTool ...
```
The part acc.bettermarks.com should be configurable

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+    '''
+
+    # Create Course
+    courseId = createCourse(self, courseDataBuilder(self))
```
Is the order right here? Can a Pupil create a course?

FelixBlauza replied:
The order is right. Pupil User will not reach this section.

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+                self.client.post(
+                    mainHost + "/r0/rooms/" + room_id + "/send/m.room.message",
+                    json=data,
+                    #name="https://matrix.niedersachsen.messenger.schule/_matrix/client/r0/rooms/" + room_id + "/send/m.room.message"
```
Please remove commented out code

FelixBlauza replied:
Removed

dimapin commented on requestsBuilder.py:
```diff
@@ -0,0 +1,548 @@
+        else:
+            return True
+
+def matrixMessenger(self):
```
Leave a comment about a status

FelixBlauza replied:
Done

Commit:
[OPS-2598] Implementation of changes requests from pull-request

- Changed requestWithoutUser: url will now use the host url for get-requests
- Changed comments

Commit:
[OPS-2598] Fixed Problems from Pull Request

Commit:
[OPS-2598] Added imports on functions.py

Commit:
[OPS-2598] Fixed Problems from Pull Request

Commit:
[OPS-2598] Fixed Problems from Pull Request

Commit:
[OPS-2598] Added search for .svg and fonts

Commit:
[OPS-2598] Editet url from bettermarks

MartinNeum committed:
Update README.md

Edited the description of the loadtest

MartinNeum committed:
Merge branch 'main' into loadtest-dev-ops-2598
